sdk/python/kfp/compiler/compiler_utils.py

```diff
@@ -274,7 +274,13 @@ def get_inputs_for_all_groups(
                             channels_to_add.pop()
                             if not channels_to_add:
                                 break
-
+            elif isinstance(channel, for_loop.Collected):
```

Comment:
From a code-readability perspective, the original if-else was channel produced by task (task_name being non-empty value) vs. channel not produced by task (task_name being empty value). Order-wise, It's better not to insert the new case in the middle of the two. And with these three cases, it's better to explicitly check each.

Reply:
Makes sense, though no longer relevant per [other comment](https://github.com/kubeflow/pipelines/pull/8631#discussion_r1092326763).